### ISSUE TITLE
Design: Investment Performance

### ISSUE THREAD

**Issue by redstreet** [redstreet/fava-investor#26]
Continuing the conversation from the [net worth work](https://github.com/redstreet/fava-investor/pull/20). Here are some thoughts. Perhaps, more than net worth with Fava already displays on the main page (which is by definition across one's entire accounts, and not restricted to investments), what we are interested in here is investment performance analysis. Refining our goals further:

# Investment Performance Analysis: Design Goals

### What do we seek to understand?

Let's understand what we may want out of the investment performance analysis component
by asking typical questions around this topic:

- What's the balance in my investment accounts? How much has it gone up/down by in the
  past?
- How much of my balance has been my original contributions, vs. returns from the
  investment, and  appreciation
- How much have I contributed over time?
- How much have I withdrawn over time?
- How is a specific account (or set of accounts) doing with respect to all the questions
  above?

### What does a mock up look like?

We want to show graphs for the following:
- show balances across time

- balance breakdown across time: show split of 1) contributions, 2) income (dividends,
  gains, interest, etc.), and 3) appreciation

- purchases and withdrawals (cumulative) across time

All these should be:
- filterable by accounts (using standard fava interface)
- across arbitrary time periods (using standard fava interface)

**Comment by redstreet** [comment]
@baartosz, I'd love to hear your thoughts. I see what you have is already headed in this direction.

**Comment by baartosz** [comment]
Thats pretty close to what I imagined.
Module name "Networth" is a bit misleading, "performance" certainly is more accurate.

**Comment by baartosz** [comment]
Queries don't seem suitable for the job. We would need to query plenty of small periods to get data for chart and that's probably going to be slow. Seems like going through all entries and classifying them would work better. What do you think?

**Comment by redstreet** [comment]
:+1: on classifying entries. Queries are probably useful for prototyping, and also as a reference for testing IMHO.

You could look at what Fava does for its main net worth chart and for the journal. Both of them have a similar problem - i.e., needing to know the balances over short periods of time.

**Comment by redstreet** [comment]
I've fleshed out the approach in the second post from top. Would love to know your thoughts. I'm excited to see the movement on this. Thank you for the pushing this forward!

**Comment by baartosz** [comment]
Do I understand correctly that "balance analyzer" is name for a tool that will show split into contributions, dividends and gains using previous tools?

Yes, it looks like good roadmap. Back to prototyping!

**Comment by redstreet** [comment]
I meant showing just balances without any splitting. So the data part would be very simple and easily analyzable and testable.

**Comment by baartosz** [comment]
Seems like that screen covers point 1. Is something missing there?
https://fava.pythonanywhere.com/example-beancount-file/account/Assets/balances/?interval=year

**Comment by redstreet** [comment]
You're right, it does, thanks for pointing this out! I tend to use the net worth chart that includes Liabilities, and hadn't thought of this one in a while.

The only thing it may be missing is to choose an arbitrary subset of the accounts through a regular expression. But this is not an often used feature.

It does seem to require a bit of navigation to get it to display that page. If you want to take a few minutes to make it the default tab within performance, that'd be cool. If you want to instead just move on to item 2, that'd be cool too.
